### PR TITLE
Update Rect intersection documentation, and rename method on Mono

### PR DESCRIPTION
**Pull request by nathanfranke**

### CODE REVIEW

**Review by aaronfranke — APPROVED** [review]
The C# changes are good, and the doc changes also look good.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs [hunk 1]
```diff
@@ -74,10 +74,11 @@ public Rect2 Abs()
 
         /// <summary>
         /// Returns the intersection of this Rect2 and `b`.
+        /// If the rectangles do not intersect, an empty Rect2 is returned.
         /// </summary>
         /// <param name="b">The other rect.</param>
         /// <returns>The clipped rect.</returns>
-        public Rect2 Clip(Rect2 b)
```
Comment (madmiraal) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs:80]: The `<returns>` comment needs to be updated too.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2i.cs [hunk 2]
```diff
@@ -69,10 +69,11 @@ public Rect2i Abs()
 
         /// <summary>
         /// Returns the intersection of this Rect2i and `b`.
+        /// If the rectangles do not intersect, an empty Rect2i is returned.
         /// </summary>
         /// <param name="b">The other rect.</param>
         /// <returns>The clipped rect.</returns>
-        public Rect2i Clip(Rect2i b)
```
Comment (madmiraal) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2i.cs:75]: Ditto.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs (hunk 1 above)
Comment (nathanfranke) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs:80]: Changed to `The clipped rect, or an empty rect if they do not intersect.`

Comment (madmiraal) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs:80]: It shouldn't refer to clipped. It should refer to the intersection between the two rectangles:
> The intersection of this Rect2 and 'b', or an empty Rect2 if they do not intersect.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2i.cs (hunk 2 above)
Comment (madmiraal) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2i.cs:75]: Similarly:
> The intersection of this Rect2i and 'b', or an empty Rect2i if they do not intersect.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs (hunk 1 above)
Comment (nathanfranke) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Rect2.cs:80]: Changed them to `The intersection between the rects, or an empty rect if they do not intersect`, is that good?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**